src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs

```diff
@@ -3,41 +3,53 @@
 using System.Net.Sockets;
 using Raven.Abstractions.Data;
 using Raven.Abstractions.Replication;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Logging;
 
 namespace Raven.Server.Documents.Replication
 {
+	//TODO : add code to create outgoing connections
 	public class DocumentReplicationLoader : IDisposable
 	{
 		private readonly DocumentDatabase _database;
 
 		private readonly ConcurrentDictionary<ReplicationDestination,OutgoingReplicationHandler> _outgoing = new ConcurrentDictionary<ReplicationDestination, OutgoingReplicationHandler>();
-		private readonly ConcurrentDictionary<TcpConnectionHeaderMessage, IncomingReplicationHandler> _incoming = new ConcurrentDictionary<TcpConnectionHeaderMessage, IncomingReplicationHandler>();
+		private readonly ConcurrentDictionary<IncomingConnectionInfo, IncomingReplicationHandler> _incoming = new ConcurrentDictionary<IncomingConnectionInfo, IncomingReplicationHandler>();
+		private readonly Logger _log;
 
 		public DocumentReplicationLoader(DocumentDatabase database)
 		{
 			_database = database;
+			_log = _database.LoggerSetup.GetLogger<DocumentReplicationLoader>(_database.Name);
 		}
 
-		public void AcceptIncomingConnection(TcpConnectionHeaderMessage incomingMessageHeader, NetworkStream tcpStream)
+		public void AcceptIncomingConnection(TcpConnectionHeaderMessage incomingMessageHeader, 
```

Comment:
You just accepted a new connection, and you are ignoring it because there is a duplicate connection from the same source?
That is not nice to do.

You need to actually do something here to handle this scenario